examples/all-clusters-app/all-clusters-common/src/operational-state-delegates.cpp

```diff
@@ -0,0 +1,184 @@
+    GenericOperationalState(to_underlying(OperationalStateEnum::kRunning)),
+    GenericOperationalState(to_underlying(OperationalStateEnum::kPaused)),
+    GenericOperationalState(to_underlying(OperationalStateEnum::kError)),
+};
+
+/**
+ * Enquriy Table of Phase List
+ * Note: User Define
+ */
+static const GenericOperationalPhase opPhaseList[] = {
+    /**
+     * Phase List is null
+     */
+    GenericOperationalPhase(DataModel::Nullable<CharSpan>()),
+};
+
+/**
+ * Enquriy Table of Operational State Delegate
+ * Note: User Define
+ */
+static OperationalStateDelegate opStateDelegate(Clusters::OperationalState::kDemoEndpointId, Clusters::OperationalState::Id,
+                                                GenericOperationalState(to_underlying(OperationalStateEnum::kStopped)),
+                                                GenericOperationalError(to_underlying(ErrorStateEnum::kNoError)),
+                                                Span<const GenericOperationalState>(opStateList),
+                                                Span<const GenericOperationalPhase>(opPhaseList));
+
+/**
+ * Enquriy Table of Operational State Cluster and alias Cluter Delegate corresponding to endpointId and clusterId
+ * Note: User Define
+ */
+constexpr DelegatesEnquiryTable kDelegatesEnquiryTable[] = {
+    // EndpointId, ClusterId, Delegate
+    { Clusters::OperationalState::kDemoEndpointId, Clusters::OperationalState::Id, &opStateDelegate },
+};
+
+/**
+ * Get the pointer of target delegate for target endpoint and cluster
+ * @param[in] aEndpiontId The endpointId
+ * @param[in] aClusterID  The clusterId
+ * @return the pointer of target delegate
+ */
+Delegate * getGenericDelegateTable(EndpointId aEndpointId, ClusterId aClusterId)
+{
+    for (size_t i = 0; i < ArraySize(kDelegatesEnquiryTable); ++i)
+    {
+        if (kDelegatesEnquiryTable[i].mEndpointId == aEndpointId && kDelegatesEnquiryTable[i].mClusterId == aClusterId)
+        {
+            return kDelegatesEnquiryTable[i].pItems;
+        }
+    }
+    return nullptr;
+}
+
+// @brief Instance getter for the default global delegate for operational state alias cluster
+// The delegate API assumes there will be separate delegate objects for each cluster instance.
+// (i.e. each separate operational state cluster derivation, on each separate endpoint)
+// @note This API should always be called prior to using the delegate and the return pointer should never be cached.
+// @return Default global delegate instance.
+Delegate * GetOperationalStateDelegate(EndpointId endpointId, ClusterId clusterId)
+{
+    return getGenericDelegateTable(endpointId, clusterId);
+}
+
+/**
+ * Log OperationalError Event
+ * @param[in] aEndpointId Target endpointId to log event
+ * @param[in] err Set the operational error to event.
+ * @return true: send event success; fail : send event fail.
+ */
+bool LogOperationalErrorEvent(EndpointId aEndpointId, const GenericOperationalError & err)
```

Comment:
Shouldn't this be called when the operational error actually changes?  I feel like above we are not doing that, so this example is setting up people to fail if they copy it...

I strongly suggest taking this out, and filing a followup issue to emit the right events and log the right attribute changes when operational error changes.   Ideally that logic would live in a cluster API, not in applications.

Reply:
Has take it out.